### PR TITLE
Remove worker,server,cache

### LINKED ISSUE #102: Fix cwd when loading server
I've noticed that deoplete-jedi will use Neovim's `cwd` when loading the server, which means that this depends on `:lcd` and when the server gets started (when doing it lazily, e.g. when entering insert mode).

When using `pyenv`, this might result in different `python` interpreters being used.

While this can be powerful in the context of restarting deoplete-jedi with another interpreter, it is rather confusing in its current state.

1. A first step here might be to allow for configuration of the `python` prog (https://github.com/zchee/deoplete-jedi/blob/cdbe0902d646c37e68d0b2967b445c993b78cd32/rplugin/python3/deoplete/sources/deoplete_jedi/server.py#L464).
This should be looked at when restarting then, too - allowing to change it without restarting Neovim (https://github.com/zchee/deoplete-jedi/blob/cdbe0902d646c37e68d0b2967b445c993b78cd32/rplugin/python3/deoplete/sources/deoplete_jedi/server.py#L500).

2. Is there a way to get Neovim's startup directory later?  This could be used then to change into before running the `python` prog.  This would make the automatic detection (in the case of `pyenv` or other tools using the current working directory) more deterministic.

### PR DESCRIPTION
**Pull request by Shougo**
Fix #102 #103 #150 #151 #155 #162 #163 #175 #190 #191 #184

Continues #184 

Please test the branch.
I think it is better.

@zchee @tweekmonster  Can you review it?

### CODE REVIEW

File: rplugin/python3/deoplete/sources/deoplete_jedi.py [hunk 1]
```diff
@@ -156,8 +156,12 @@ def gather_candidates(self, context):
...
                    line, col, filename, modified)
 
-        script = self.get_script(source, line, col, filename,
-                                 environment=self._env)
+        try:
+            script = self.get_script(source, line, col, filename,
+                                     environment=self._env)
+        except ValueError:
+            # Ignore ValueError
+            return []
```
Comment (blueyed) [inline rplugin/python3/deoplete/sources/deoplete_jedi.py]: Why?
What does it ignore?

File: rplugin/python3/deoplete/sources/deoplete_jedi.py [hunk 2]
```diff
@@ -159,12 +159,11 @@ def gather_candidates(self, context):
         try:
             script = self.get_script(source, line, col, filename,
                                      environment=self._env)
+            completions = self.get_completions(script)
         except ValueError:
             # Ignore ValueError
             return []
 
-        completions = self.get_completions(script)
-
```
Comment (blueyed) [inline rplugin/python3/deoplete/sources/deoplete_jedi.py]: Same question as before.

Comment (Shougo) [reply rplugin/python3/deoplete/sources/deoplete_jedi.py]: ValueError(invalid column) is generated from jedi.
It seems from asynchronous.

Comment (Shougo) [reply rplugin/python3/deoplete/sources/deoplete_jedi.py]: The fix is wellcome.  I think the better check is needed.

**Comment by Shougo** [comment]
I will improve the implementation and merge it later.

**Comment by Shougo** [comment]
@blueyed I have added `g:deoplete#sources#jedi#enable_typeinfo` option.
You can disable the option.  It will be faster.  But you cannot get the type information.

File: rplugin/python3/deoplete/sources/deoplete_jedi.py [hunk 3]
```diff
@@ -20,6 +20,9 @@
 sys.path.insert(0, jedi_path)
 import jedi  # noqa: E402
 
+# Todo: workaround for Python 3.7
+jedi.api.environment._SUPPORTED_PYTHONS.insert(0, '3.7')
```
Comment (blueyed) [inline rplugin/python3/deoplete/sources/deoplete_jedi.py]: This should handle the case for when it gets added (https://github.com/davidhalter/jedi/pull/1217), i.e. only add it if not there yet (although we control this by updating the submodule ourselves currently, but might be forgotten also anyway).

**Comment by Shougo** [comment]
I will merge it in the weekend.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**